src/index.js

```diff
@@ -1,32 +1,48 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const BASE_API = "https://rickandmortyapi.com/api/character/",
+  xhr = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+const fetchData = url => {
+  xhr.open('GET', url, false)
+  xhr.responseType = 'json'
+  return new Promise((res, rej) => {
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          const response = JSON.parse(xhr.responseText)
+          return res(response)
+        } else {
+          return rej(new Error('SORRY! ERROR HAS OCURRED :('))
+        }
+      }
     }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
-
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+    xhr.send()
+  })
+}
+
+const searchData  = async url => {
+  try {
+    console.log('L O A D I N G')
+    const world = await fetchData(url)
+
+    console.log('L O A D I N G . . .')
+    const chapters = await fetchData(`${BASE_API}${world.results[0].id}`)
+
+    console.log('L O A D I N G . . . . . .')
+    const characters = await fetchData(`${chapters.origin.url}`)
+
+    console.log(' ')
+    console.log('------------------------------')
+    console.log(' ')
+
+    console.log(`1. Count characters: ${world.info.count}`)
+    console.log(`2. First character: ${chapters.name}`)
+    console.log(`3. Dimensions: ${characters.dimension}`)
+
+    }catch(error) {
```

Comment:
Hay que tener cuidado con el indentado, si bien no es un factor que afecte la lógica de tu problema, si es un factor importante para la lectura del código. 

```suggestion
    }
    catch(error) {
```